src/Parser/Druid/Balance/Modules/Core/GlobalCooldown.js

```diff
@@ -0,0 +1,56 @@
+import SPELLS from 'common/SPELLS';
+import CoreGlobalCooldown from 'Parser/Core/Modules/GlobalCooldown';
+import Channeling from 'Parser/Core/Modules/Channeling';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import Abilities from '../Features/Abilities';
+import Haste from './Haste';
+
+/**
+ * The talent Starlord reduces GCD and cast time of empowered Lunar Strikes and Solar Wraths by 20%. 
+ * Since Solar Wrath cast time == GCD the GCD needs to be reduced.
+ */
+class GlobalCooldown extends CoreGlobalCooldown {
+  static dependencies = {
+    abilities: Abilities,
+    haste: Haste,
+    combatants: Combatants,
+    channeling: Channeling,
+  };
+
+  isStarlordBuffedSolarWrath(event) {
+  	return event.ability.guid === SPELLS.SOLAR_WRATH_MOONKIN.id && this.combatants.selected.hasTalent(SPELLS.STARLORD_TALENT.id) && this.combatants.selected.hasBuff(SPELLS.SOLAR_EMP_BUFF.id);
+  }
+
+  on_byPlayer_begincast(event) {
```

Comment:
My suggestion was to override the `getCurrentGlobalCooldown` method and making it return a 20% reduced GCD when you have the buff and it's an applicable event, would that not suffice?

It will take some more time to review this as it is since it's harder to grasp what behavior is being changed. I think this might break the channel/gcd time behavior but I haven't reviewed the base class to make sure.

Reply:
yeah that would probably have been simpler, I'll take a look at that.